hail/python/hailtop/aiotools/fs.py

```diff
@@ -822,6 +839,18 @@ def _get_fs(self, url):
 
         return fs
 
+    async def read(self, url: str) -> bytes:
+        fs = self._get_fs(url)
+        return await fs.read(url)
+
+    async def read_from(self, url: str, start: int) -> bytes:
+        fs = self._get_fs(url)
+        return await fs.read_from(url, start)
+
+    async def read_range(self, url: str, start: int, end: int) -> bytes:
+        fs = self._get_fs(url)
+        return await fs.read_range(url, start, end)
+
```

Comment:
I don't think these need to be here. The implementations inherited from AsyncFS should work just fine, right?

Reply:
I think you need to grab the appropriate file system first. This is the RouterFS. That's what all the other methods do.